admin/services/prepare-check.service.js

```diff
@@ -0,0 +1,63 @@
+'use strict'
+
+const moment = require('moment')
+const R = require('ramda')
+const redisService = require('./redis-cache.service')
+
+const service = {
+  /**
+   * @param {[{object}]} checks the pupil checks to prepare
+   * @returns {Promise<void>}
+   */
+  prepareChecks: async (checks) => {
+    if (!Array.isArray(checks)) {
+      throw new Error('checks is not an array')
+    }
+    const batch = []
+    for (let index = 0; index < checks.length; index++) {
+      const check = checks[index]
+      const preparedCheck = constructPreparedCheck(check)
+      const cacheKey = buildKey(check.schoolPin, check.pupilPin)
+      batch.push({
+        key: cacheKey,
+        value: preparedCheck
+      })
+      const ttl = secondsBetweenNowAnd4pm()
+      return redisService.addBatch(batch, ttl)
+    }
+  }
+}
+
+function secondsBetweenNowAnd4pm () {
+  const a = moment()
+  const b = moment().startOf('16:00')
+  return a.diff(b, 'seconds')
+}
+
+/**
+ * Return an entity suitable for inserting into the `preparedCheck` table
+ * @param {object} check
+ * @return {object}
+ */
+function constructPreparedCheck (check) {
+  const entity = {
+    checkCode: check.pupil.checkCode,
+    config: check.config,
+    createdAt: new Date(),
+    pinExpiresAt: moment(check.pupil.pinExpiresAt).toDate(),
+    pupil: R.omit(['id', 'checkFormAllocationId', 'pinExpiresAt'], check.pupil),
+    pupilId: check.pupil.id,
+    questions: check.questions,
+    school: check.school,
+    schoolId: check.school.id,
+    tokens: check.tokens,
+    updatedAt: new Date()
```

Comment:
Would be good to have the 'valid from' information.  We have `pinExpiresAt`, so perhaps call it `pinValidFrom`?